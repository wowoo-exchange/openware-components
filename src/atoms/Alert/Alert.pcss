.cr-alert {
  align-items: center;
  background: var(--alert-background-color);
  border-radius: calc(var(--gap) * 0.5);
  box-sizing: border-box;
  box-shadow: 0 10px 15px rgba(0, 0, 0, 0.25);
  display: flex;
  min-height: calc(var(--gap) * 9.3);
  justify-content: space-between;
  padding: 0 calc(var(--gap) * 3.16) 0 calc(var(--gap) * 10);
  position: relative;
  text-align: right;
  width: calc(var(--gap) * 50);
  margin-top: calc(var(--gap) * 0.6);

  &__title,
  &__description {
    background: transparent;
    line-height: 1em;
    margin: 0;
    color: var(--alert-description-color);
  }

  &-text {
    background: transparent;
    width: 100%;
  }

  &__title {
    line-height: calc(var(--gap) * 2.33);
    margin-bottom: 5px;
  }

  &--success {
    border: 1px solid var(--open-orders-order-buy);

    .cr-alert__title {
      color: var(--open-orders-order-buy);
    }
  }

  &--error {
    border: 1px solid var(--open-orders-order-sell);

    .cr-alert__title {
      color: var(--open-orders-order-sell);
    }
  }
}
