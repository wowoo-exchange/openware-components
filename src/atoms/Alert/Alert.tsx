import cx from 'classnames';
import * as React from 'react';
import { AlertIcon } from '../AlertIcon/AlertIcon';

interface AlertProps {
    /**
     * Additional message
     */
    description: string;
    /**
     * Alert message
     */
    title: string;
    /**
     * Type of message
     */
    type: 'success' | 'error';
}

const Alert: React.FunctionComponent<AlertProps> = props => {
    const { description, type, title } = props;
    const className = cx('cr-alert', {
        'cr-alert--error': type === 'error',
        'cr-alert--success': type === 'success',
    });

    return (
        <div className={className}>
            <AlertIcon type={type} />
            <div className="cr-alert-text">
                <p className="cr-alert__title">{title}</p>
                <p className="cr-alert__description">{description}</p>
            </div>
        </div>
    );
};

export {
    Alert,
    AlertProps,
};
