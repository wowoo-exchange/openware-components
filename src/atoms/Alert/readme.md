```js
<Alert description="30.052 BTC" title="Successful deposit" type="success" />
<Alert description="Internal server error" title="Withdraw failure" type="error" />
```