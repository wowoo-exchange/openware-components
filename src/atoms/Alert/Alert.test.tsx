import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import { Alert, AlertProps } from './Alert';

const defaultProps: AlertProps = {
    description: '34.233 BTC',
    title: 'Successful deposit',
    type: 'success',
};

const setup = (props: Partial<AlertProps> = {}) =>
    shallow(<Alert {...{ ...defaultProps, ...props }} />);

describe('Alert', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = setup();
    });

    it('should render', () => {
       expect(wrapper).toMatchSnapshot();
    });

    it('should render success message', () => {
        wrapper = setup({ type: 'success' });
        const container = wrapper.find('.cr-alert');
        expect(container.hasClass('cr-alert--success')).toBeTruthy();
    });

    it('should render error message', () => {
        wrapper = setup({ type: 'error' });
        const container = wrapper.find('.cr-alert');
        expect(container.hasClass('cr-alert--error')).toBeTruthy();
    });
});
