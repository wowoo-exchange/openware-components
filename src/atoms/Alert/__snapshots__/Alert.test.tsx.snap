// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render 1`] = `
<div
  className="cr-alert cr-alert--success"
>
  <AlertIcon
    type="success"
  />
  <div
    className="cr-alert-text"
  >
    <p
      className="cr-alert__title"
    >
      Successful deposit
    </p>
    <p
      className="cr-alert__description"
    >
      34.233 BTC
    </p>
  </div>
</div>
`;
